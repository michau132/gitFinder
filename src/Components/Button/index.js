/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './style.css';

const Button = ({
  type, className, onClick, text, disabled,
}) => (
  <button
    disabled={disabled}
    type={type}
    className={classNames(className, styles.button, { [styles.disabledBtn]: disabled })}
    onClick={onClick}
  >
    {text}
  </button>
);

Button.defaultProps = {
  className: null,
  onClick: null,
  disabled: false,
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
};

export default Button;
