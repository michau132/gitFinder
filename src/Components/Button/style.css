.button {
  text-transform: uppercase;
  padding: 5px 10px;
  background: #fff;
  border: 2px solid #7619FF;
  color: #7619FF;
  transition: 0.2s all;
  margin-left: 10px;
}

.button:hover {
  background: #43d1af;
  color: #fff;
}

.button:disabled {
  background-color: red;
}
