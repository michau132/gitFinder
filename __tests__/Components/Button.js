import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../../src/Components/Button';

describe('testing Button Component', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = {
      onClick: jest.fn(),
      type: 'button',
      className: 'dangerButton',
      disabled: false,
      text: 'open',
    };
    wrapper = shallow(<Button {...props} />);
  });
  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('calls onClick function when button is not disabled', () => {
    wrapper.simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });

  test('checking type prop', () => {
    expect(wrapper.prop('type')).toBe('button');
  });

  test('checking className prop', () => {
    expect(wrapper.prop('className')).toBe('dangerButton');
  });

  test('checking disabled prop', () => {
    expect(wrapper.prop('disabled')).toBeFalsy();
  });

  test('checking disabled prop', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper.prop('disabled')).toBeTruthy();
    expect(props.onClick).toHaveBeenCalledTimes(0);
  });
});
