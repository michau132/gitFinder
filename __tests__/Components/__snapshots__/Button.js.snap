// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button Component matching snapshot 1`] = `
<button
  className="dangerButton"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  open
</button>
`;
